### PR TITLE
Bug 1353105, Bug 1353114 - Main summary updates

### PR DESCRIPTION
**Pull request by fbertsch**

### CODE REVIEW

File: src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala [hunk 1]
```diff
@@ -12,9 +12,19 @@ case class BooleanScalar(keyed: Boolean) extends ScalarDefinition
...
+
   // mock[io.Source] wasn't working with scalamock, so now we'll just use the function
   protected val getURL: (String, String) => scala.io.BufferedSource = Source.fromURL
 
+  def getParquetFriendlyScalarName(scalarName: String, processName: String): String = {
+    // Scalar group and probe names can contain dots ('.'). But we don't
+    // want them to get into the Parquet column names, so replace them with
+    // underscores ('_'). Additionally, to prevent name clashing, we prefix
+    // all scalars.
+    ScalarColumnNamePrefix + '_' + (processName + '_' + scalarName).replace('.', '_')
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala]: Should we also force these names to lowercase and replace any other not-parquet-friendly characters?

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 2]
```diff
@@ -570,7 +571,29 @@ object MainSummaryView {
     StructField("dom_ipc_process_count", IntegerType, nullable = true) // dom.ipc.processCount
   ))
 
-  def buildSchema: StructType = {
+  def buildScalarSchema(scalarDefinitions: List[(String, ScalarDefinition)]): List[StructField] = {
+    //TODO: optin/optout, other processes
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Please file a follow-up bug or issue and remove this TODO

File: src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala [hunk 3]
```diff
@@ -683,23 +706,19 @@ object MainSummaryView {
...
       StructField("events", ArrayType(Events.buildEventSchema, containsNull = false), nullable = true), // payload.processes.parent.events
 
       // bug 1339655
       StructField("ssl_handshake_result_success", IntegerType, nullable = true),
       StructField("ssl_handshake_result_failure", IntegerType, nullable = true),
-      StructField("ssl_handshake_result", MapType(StringType, IntegerType), nullable = true) // SSL_HANDSHAKE_RESULT
-    ))
+      StructField("ssl_handshake_result", MapType(StringType, IntegerType), nullable = true), // SSL_HANDSHAKE_RESULT
+
+      // simple measures - bug 1353114
```
Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/views/MainSummaryView.scala]: Please add the full `payload.simpleMeasurements.*` path to this comment for reference back to the raw pings.

File: src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala [hunk 4]
```diff
@@ -676,13 +686,19 @@ class MainSummaryViewTest extends FlatSpec with Matchers{
...
             "user_prefs"                        -> null,
-            "max_concurrent_tab_count"          -> null,
-            "tab_open_event_count"              -> null,
-            "max_concurrent_window_count"       -> null,
-            "window_open_event_count"           -> null,
-            "total_uri_count"                   -> null,
-            "unfiltered_uri_count"              -> null,
-            "unique_domains_count"              -> null
+            "active_ticks"                      -> 17354,
+            "main"				                -> 199,
```
Comment (mreid-moz) [inline src/test/scala/com/mozilla/telemetry/MainSummaryViewTest.scala]: nit: Please fix the indentation/alignment here

File: src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala (hunk 1 above)
Comment (fbertsch) [inline src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala]: I can't find a canonical list of parquet-friendly chars so I think we can just lazily evaluate non-friendly chars in the future when we run into them.

Comment (mreid-moz) [inline src/main/scala/com/mozilla/telemetry/scalars/Scalars.scala]: Ok, sounds good.

**Comment by fbertsch** [comment]
@mreid-moz RFAL

File: docs/MainSummary.md [hunk 5]
```diff
@@ -192,7 +185,20 @@ root
  |    |    |-- map_values: map (nullable = true)
  |    |    |    |-- key: string
  |    |    |    |-- value: string
+ |-- activeTicks: long (nullable = true)
```
Comment (mreid-moz) [inline docs/MainSummary.md]: This set of `simpleMeasurements` fields should use `snake_case` naming - `active_ticks`, `first_paint`, `session_restored`, `total_time`

File: docs/MainSummary.md [hunk 6]
```diff
@@ -192,7 +185,20 @@ root
...
+ |-- sessionRestored: long (nullable = true)
+ |-- totalTime: long (nullable = true)
+ <SCALARS>
 ```
+
+### Scalars
+Parent scalars are automatically added to the dataset, and as such, there is no longer a fixed schema.
+They are ordered by the name of the scalar, which is composed of <namespace>.<scalar_name>. The full
+schema name is scalar_<process>_<namespace>_<scalar>, where currently only parent process scalars are
+available.
```
Comment (mreid-moz) [inline docs/MainSummary.md]: Can you add a specific example of one of the scalar fields here?

**Comment by fbertsch** [comment]
@mreid-moz updated docs and bumped the version.

**Comment by fbertsch** [comment]
@mreid-moz the schema was different:
1. `sample_id` and `is_wow64` changed places
2. several fields have `nullable=false` (e.g. `submission_date_s3`, `timestamp`, some structs, etc.)
3. `attribution` was missing

**Comment by fbertsch** [comment]
@mreid-moz, today's the day (https://youtu.be/-OZUc5ZsTUY?t=49)! Fixed the doc versions :)

**Pull request merged**